File: model/src/main/java/io/github/runtimemodels/chazm/entity/CharacteristicFactory.java
package io.github.runtimemodels.chazm.entity;

import io.github.runtimemodels.chazm.id.UniqueId;

/**
 * The {@linkplain CharacteristicFactory} interface defines the APIs for constructing {@linkplain Characteristic}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface CharacteristicFactory {

    /**
     * Constructs a {@linkplain Characteristic}.
     *
     * @param id the {@linkplain UniqueId} that represents the {@linkplain Characteristic}.
     * @return a {@linkplain Characteristic}.
     */
    Characteristic buildCharacteristic(UniqueId<Characteristic> id);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/factory/CharacteristicFactory.kt
package runtimemodels.chazm.model.factory

import runtimemodels.chazm.api.entity.Characteristic
import runtimemodels.chazm.api.id.CharacteristicId
import runtimemodels.chazm.api.id.UniqueId

/**
 * The [CharacteristicFactory] interface defines the APIs for constructing [Characteristic]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface CharacteristicFactory {

    /**
     * Constructs a [Characteristic].
     *
     * @param id the [UniqueId] that represents the [Characteristic].
     * @return a [Characteristic].
     */
    fun buildCharacteristic(id: CharacteristicId): Characteristic

}
